evap/staff/importers/json.py

```diff
@@ -0,0 +1,445 @@
+            if not email:
+                self.statistics.warnings.append(
+                    WarningMessage(
+                        obj=f"Contributor {entry['christianname']} {entry['name']}", message="No email defined"
+                    )
+                )
+            else:
+                user_profile, __, changes = update_or_create_with_changes(
+                    UserProfile,
+                    email=email,
+                    defaults={
+                        "last_name": entry["name"],
+                        "first_name_given": entry["christianname"],
+                        "title": entry["titlefront"],
+                    },
+                )
+                if changes:
+                    self._create_name_change_from_changes(user_profile, changes)
+
+                self.students_by_gguid[entry["gguid"]] = user_profile
+
+    def _import_course(self, data: ImportEvent, course_type: CourseType | None = None) -> Course:
+        course_type = self._get_course_type(data["type"]) if course_type is None else course_type
+        responsibles = self._get_user_profiles(data["lecturers"])
+        responsibles = self._remove_non_responsible_users(responsibles)
+        responsibles = self._get_users_with_longest_title(responsibles)
+        if not data["title_en"]:
+            data["title_en"] = data["title"]
+        course, created, changes = update_or_create_with_changes(
+            Course,
+            semester=self.semester,
+            cms_id=data["gguid"],
+            defaults={"name_de": data["title"], "name_en": data["title_en"], "type": course_type},
+        )
+        course.responsibles.set(responsibles)
+
+        if created:
+            self.statistics.new_courses.append(course)
+        elif changes:
+            self.statistics.updated_courses.append(course)
+
+        self.courses_by_gguid[data["gguid"]] = course
+
+        return course
+
+    def _import_course_programs(self, course: Course, data: ImportEvent) -> None:
+        programs = [
+            self._get_program(c["cprid"]) for c in data["courses"] if c["cprid"] not in settings.IGNORE_PROGRAMS
+        ]
+        course.programs.add(*programs)
+
+    def _import_course_from_unused_exam(self, data: ImportEvent) -> Course | None:
+        split_title = data["title"].split(":", 1)
+        if len(split_title) < 2:
+            return None
+        prefix = split_title[0].strip()
+
+        try:
+            course_type = CourseType.objects.get(import_names__contains=[prefix])
+        except CourseType.DoesNotExist:
+            return None
+
+        data["title"] = split_title[1].strip()
+        data["title_en"] = data["title_en"].split(":", 1)[1].strip() if ":" in data["title_en"] else data["title_en"]
+
+        return self._import_course(data, course_type)
+
+    # pylint: disable=too-many-locals
+    def _import_evaluation(self, course: Course, data: ImportEvent) -> Evaluation:
+        if "appointments" not in data:
```

Comment:
To avoid getting max of empty iterable:
```suggestion
        if "appointments" not in data or not data["appointments"]:
```